### PR TITLE
lightning: fix panic when table name in source file and target cluster is different

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31771

Problem Summary:

Lightning will panic when table name in source file and target cluste…

### What is changed and how it works?

 Fix panic when table name in source file and target cluster is different.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/tidb.go [hunk 1]
```diff
@@ -272,10 +272,10 @@ func LoadSchemaInfo(
 			tableInfo := &checkpoints.TidbTableInfo{
 				ID:   tblInfo.ID,
 				DB:   schema.Name,
-				Name: tableName,
+				Name: tbl.Name,
```
Comment (glorv) [inline br/pkg/lightning/restore/tidb.go:275]: Please add a log here to describe why we need to use `tbl.Name` instead of `tableName`

**Review by glorv — APPROVED** [review]
LGTM

**Comment by gozssky** [comment]
ping @kennytm

**Review by kennytm — APPROVED** [review]

**Pull request merged**